lib/katello/tasks/clean_backend_objects.rake

```diff
@@ -20,7 +20,7 @@ namespace :katello do
           system.system_activation_keys.destroy_all
           system.system_host_collections.destroy_all
           system.system_errata.destroy_all
-          system.delete
+          system.destroy!
```

Comment:
I'm okay with this change, the reason it was .delete() before was due to the AR hooks that were handling the orchestration.  That no longer is the case.  However if we do change it to destroy should remove the previous 3 lines as they are now handled by AR relationships.